src/context-selection/org-unit-selector-bar-item/org-unit-selector-bar-item.js

```diff
@@ -106,12 +131,36 @@ export default function OrganisationUnitSetSelectorBarItem() {
                                             // Not sure why this is necessary, but when not done,
                                             // it causes bugs in the UI
                                             e.stopPropagation()
-                                            setOrgUnitId(id)
-                                            setOrgUnitOpen(false)
+                                            if (
+                                                assignedOrgUnits?.includes(id)
+                                            ) {
+                                                setOrgUnitId(id)
+                                                setOrgUnitOpen(false)
+                                            }
+                                        }}
+                                        renderNodeLabel={({ node, label }) => {
+                                            return assignedOrgUnits?.includes(
+                                                node?.id
+                                            ) ? (
+                                                label
+                                            ) : (
+                                                <UnclickableLabel
+                                                    label={label}
+                                                />
+                                            )
                                         }}
                                     />
                                 )}
                         </div>
+                        <Divider margin="0" />
+                        <div className={css.labelContentContainer}>
+                            <IconBlock16 />
+                            <span className={css.label}>
+                                {i18n.t(
+                                    'Dataset is not assigned to this organisation unit'
+                                )}
+                            </span>
+                        </div>
```

Comment:
I found this a bit confusing until I realised that this is a "legend". I'm wondering if we should add a headline or if we should move this to the `Help` menu which contains a legend for cells as well as keyboard shortcuts already (cc @cooper-joe)

Reply:
The styling/legend here is copied from the approvals app, but I'm definitely happy to update as people want

@cooper-joe do you want us to hold off merging this until you've finalized the design? Or should I merge and then you can revisit the design later? 

This is what it looks like now:
<img width="350" alt="org_unit_restrictions" src="https://user-images.githubusercontent.com/18490902/188445829-72e03b2a-4bf5-4d2f-ae0b-102ae7558557.png">

Reply:
Ah, if this is an established pattern, then I don't mind.

> @cooper-joe do you want us to hold off merging this until you've finalized the design? Or should I merge and then you can revisit the design later?

I'm for merging this and address design changes later. It will result in the same amount of work as we would have to do now anyway, so I don't see a reason to defer merging this PR

Reply:
Agreed that it is slightly confusing when there's only a single legend. The pattern works fine in Data Approvals, but is a bit sub-optimal here. However, agreed that we should merge this PR now and revisit the design separately (if it's a priority).